### PR TITLE
add saveStack method and helpers and support Expressions

### PR DESCRIPTION
**Pull request by chregu**
Needs rokka stack expressions to be released on the backend first. But it actually is BC, expressions just don't get saved for now

### CODE REVIEW

File: src/Core/Stack.php [hunk 1]
```diff
@@ -104,6 +174,20 @@ public function getName()
     }
 
     /**
+     * @since 1.1.0
+     *
+     * @param string $name
+     *
+     * @return $this
+     */
+    public function setName($name): self
```
Comment (thePanz) [inline src/Core/Stack.php]: We're still with PHP 5.6, no return typehints :(

File: src/Image.php [hunk 2]
```diff
@@ -295,6 +297,10 @@ public function listOperations()
     /**
      * Create a stack.
      *
+     * @deprecated 2.0.0
```
Comment (thePanz) [inline src/Image.php]: I guess this line can be combined with the previous one

File: src/Core/Stack.php (hunk 1 above)
Comment (chregu) [reply src/Core/Stack.php]: fixed and test added

File: src/Image.php (hunk 2 above)
Comment (chregu) [reply src/Image.php]: true, fixed

File: src/Core/StackExpression.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+     */
+    public function getExpression()
+    {
+        return $this->expression;
+    }
+
+    /**
+     * @param string $expression
+     *
+     * @return StackExpression
```
Comment (thePanz) [inline src/Core/StackExpression.php:57]: Sometimes we have `return $this`, I am still unsure if the correct one is `@return self` as mentioned here: https://docs.phpdoc.org/references/phpdoc/types.html

Comment (chregu) [reply src/Core/StackExpression.php:57]: right, changed it, seems phpstorm does the $this thing but understands also the self thing.

**Pull request merged**